src/App.js

```diff
@@ -12,138 +12,101 @@ import People from './components/People';
-                //console.error(this.props.people, status, err.toString());
-            }
-        });
-    },
-    loadNicetiesFromMe: function(callback) {
-        $.ajax({
-            url: this.props.from_me_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.fromMe, status, err.toString());
-            }
-        });
-    },
-    loadNicetiesForMe: function(callback) {
-        $.ajax({
-            url: this.props.for_me_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.niceties, status, err.toString());
-            }
-        });
-    },
-    loadSelfInfo: function(callback) {
-        $.ajax({
-            url: this.props.self_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.niceties, status, err.toString());
-            }
-        });
-    },
-    getInitialState: function() {
-        return {
-                fromMe: [],
-                people: [],
-                niceties: [],
-                currentview: "write-niceties",
-                selfInfo: [],
-        };
-    },
-    componentDidMount: function() {
-      this.loadNicetiesFromMe((data) => {
-        this.setState({fromMe: data})
-      })
-      this.loadPeopleFromServer((data) => {
-        this.setState({people: data})
-      })
-      this.loadNicetiesForMe((data) => {
-        this.setState({niceties: data})
+class App extends React.Component {
+    constructor(props) {
+      super(props)
+      this.state = {
+        fromMe: [],
+        people: [],
+        niceties: [],
+        currentview: "write-niceties",
+        selfInfo: []
+      }
+    }
+
+    loadData = (api, dataVar) => {
+      fetch(api, {
+        headers: {'Content-Type': "application/json"},
+        cache: 'no-cache'
       })
-      this.loadSelfInfo((data) => {
-        this.setState({selfInfo: data})
+      .then(response => response.json())
+      .then(data => {
+        let newState = {}
+        newState[dataVar] = data
+        this.setState(newState)
       })
-    },
-    handleSelect: function(eventKey) {
+      .catch(err => console.log(err))
+    }
+
+    componentDidMount = () => {
+      // loadPeopleFromServer
+      this.loadData(this.props.people_api, "people")
+      // loadNicetiesFromMe
+      this.loadData(this.props.from_me_api, "fromMe")
+      // loadNicetiesForMe
+      this.loadData(this.props.for_me_api, "niceties")
+      // loadSelfInfo
+      this.loadData(this.props.self_api, "selfInfo")
```

Comment:
This looks better! Mind dropping the comments?

```suggestion
      this.loadData(this.props.people_api, "people")
      this.loadData(this.props.from_me_api, "fromMe")
      this.loadData(this.props.for_me_api, "niceties")
      this.loadData(this.props.self_api, "selfInfo")
```